### ISSUE TITLE
  AGPLucerne - Growth and tissue turnover resulted in loss of mass balance

### ISSUE THREAD

**Issue by cranju** [APSIMInitiative/ApsimX#9425]
Hi,

I was simulating a crop/pasture sequence of Lucerne - Lucerne - Lucerne - Lucerne - Canola - Wheat - Barley - Faba Beans - Canola - DP Wheat - Barley. I used SCRUM to account for Faba Beans. My goal was to simulate this rotation over a 12-year period as well as over 100 years.

However, I've encountered an issue: for some years the simulation runs successfully (2000-2011), but for others, I receive a message stating, "AGPLucerne - Growth and tissue turnover resulted in loss of mass balance." Could drought be a possible reason for this?

Additionally, since I have both wheat and dual-purpose wheat (DWP) in my crop sequence, I separated them into two categories: one as wheat and the other as DWP. This allows me to use the forage from DWP. Is this the correct approach?

I have attached my simulation file for your reference.

[Crop&pasture_rotation.zip](https://github.com/user-attachments/files/17613557/Crop.pasture_rotation.zip)

![image](https://github.com/user-attachments/assets/a8b252e8-97ba-4502-b4a5-3f2849ecf2ad)

**Comment by hol353** [comment]
@sno036 FYI.

AGPLucerne is a prototype model, not a released model. There is no guarentee that it will produce sensible outputs as it hasn't been fully validated.

I've had a quick look at your simulation and for me it doesn't give a mass balance error when I run it. A mass balance error signals there is a problem in APSIM somwhere. 

Your simulation is VERY complicated which makes debugging very hard. Can you please upload a simple simulation that shows the problem?

**Comment by sno036** [comment]
This is an internal check on mass within AgPasture - we thought we had quashed it and don't get it anymore in the usual species so probably we will not come back to address it. 

As @hol353 mentioned, this is not really a validated variant of AgPasture and given that the PMF Lucerne is on the way and will be validated you would probably be best to move to that model - particularly as the germination/establishment part of AgPasture is highly simplistic. If you really want to persist with AGPLucerne then you could try doing simpler simulations to begin with, trying different cutting biomass residuals to see if the error goes away.

**Comment by cranju** [comment]
Thanks @peter-devoil @hol353 @sno036. The issue persisted when I attempted to simulate from 1924 to 2023 for both Lucerne and Red Clover (I had another system with red clover and dual-purpose crops). However, just this morning, I tried changing the soil from Wagga to a different region, and the simulation ran perfectly for the entire duration for both the systems. It seems that the pastures were sensitive to the soil characteristics I was using.

For now, we will proceed with this model and calibrate and validate it using the experimental data from the project. If PMF Lucerne becomes available during the project, we can definitely give it a try.

I also had some additional questions regarding how APSIM handles crop residue when transitioning between plants/pastures in a crop rotation. The Rotation Manager includes a "fallow" or "prepForCrop" node for transitioning between crops, but there is still crop stubble left after harvesting. If this residue isn't accounted for, it could influence the growth of subsequent crops or pastures. So, how does APSIM manage the effects of cultivation with residue in crop/pasture rotations? For pastures, I am currently killing them at the start of the simulation—does this affect the model’s treatment of residue?

For your reference, i am attaching a simplified version of the crop rotation I am doing.
[Crop&pasture_rotation_V2.zip](https://github.com/user-attachments/files/17639609/Crop.pasture_rotation_V2.zip)


@mattharrison @KeLiu7 FYI

**Comment by mattharrison** [comment]
Please check your usernames. Pretty sure you are thinking of another Matt...

**Comment by peter-devoil** [comment]
Apsim doesn't "do" anything to residues - they're just moved from one pool to another. Left alone, it will:

- At end_crop (not necessarily harvest) - transfer remaining residues (leaf & stem) to Surface Organic Matter (SOM) pools, and roots to soilN's fresh organic matter pool.

- Daily, some residues are transferred from SOM to roots via decomposition (a function of temperature, moisture and C:N ratio)

In some perverse situations (eg sugar cane, north qld), there will be more SOM accumulating from the crop than can decay over the crop cycle, and the residue pool increases forever, eventually leading to plants trying to grow their way through 100t/ha of SOM. Often, the residues will decay over a year - so it's possible to ignore it and there won't be an obvious problem.

So to mimic high production farming systems, we'll likely need to till and incorporate some of those residues  - we do this via a tillage operation, specifying the  amount of residue to incorporate, and the depth to incorporate. Which amount, and when, depends on your farming system. In minimum till systems we usually leave the residues on the surface, and incorporate them at the sowing of the next crop. Alternatively we could burn them at harvest (by specifying a depth of 0mm, in which case it disappears), or use a primary tillage implement that incorporates (eg 30%) to a deep depth. Again, it's up to the user to describe their system.

There's a tillage method in the Management toolbox  that shows how to make this happen, you could use this in different ways, eg when the crop is sown, or shortly before:

![image](https://github.com/user-attachments/assets/f0e7fd19-1784-4c05-b1bf-99d5c14b13d7)

The soil module used to modify its surface roughness (Curve Number) for some time after tillage events - I don't think that feature made it to nextgen.

**Comment by cranju** [comment]
Thanks @peter-devoil. I will try incorporating your suggestions and let you know how it goes.

**Comment by sno036** [comment]
@cranju  - is this still live?

**Comment by cranju** [comment]
@sno036 yes, i ran for my original site. But I realised the issue is still there. I was recently trying to build my simulation across different agroecological regions and surprisingly I found we still have this issue for some met files and no issue for other met files.

**Comment by sno036** [comment]
This is a very stale (perhaps fossiliaed!) issue. Please attach a simple simulation - meaning single year, single crop, simple management - in the current version of APSIM that replicates this issue. I'll look at it when I can.
